analysis/type.go

```diff
@@ -99,6 +100,56 @@ func (a *DefaultAnalyzer) Analyze(input []byte) TokenStream {
 
 var ErrInvalidDateTime = fmt.Errorf("unable to parse datetime with any of the layouts")
 
+const UnixSecs = "unix_sec"
+const UnixMilliSecs = "unix_milli"
+const UnixMicroSecs = "unix_micro"
+const UnixNanoSecs = "unix_nano"
+
+type TimestampBounds struct {
+	Min int64
+	Max int64
+}
+
+var UnixTimestampFormats = map[string]*TimestampBounds{
+	UnixSecs: {
+		Min: math.MinInt64 / 1000000000,
+		Max: math.MaxInt64 / 1000000000,
+	},
+	UnixMilliSecs: {
+		Min: math.MinInt64 / 1000000,
+		Max: math.MaxInt64 / 1000000,
+	},
+	UnixMicroSecs: {
+		Min: math.MinInt64 / 1000,
+		Max: math.MaxInt64 / 1000,
+	},
+	UnixNanoSecs: {
+		Min: math.MinInt64,
+		Max: math.MaxInt64,
+	},
+}
+
+func convertTimestamp(timestamp int64, format string) int64 {
```

Comment:
i'm thinking perhaps using a switch case each of which modify a common return value timestamp and having a single return timestamp at the end looks better.